libs/langchain/langchain/embeddings/cache.py

```diff
@@ -111,12 +114,14 @@ def embed_documents(self, texts: List[str]) -> List[List[float]]:
         vectors: List[Union[List[float], None]] = self.document_embedding_store.mget(
             texts
         )
-        missing_indices: List[int] = [
+        all_missing_indices: List[int] = [
             i for i, vector in enumerate(vectors) if vector is None
         ]
-        missing_texts = [texts[i] for i in missing_indices]
 
-        if missing_texts:
+        while all_missing_indices:
```

Comment:
Could you re-use this implementation: https://github.com/langchain-ai/langchain/blob/main/libs/core/langchain_core/utils/iter.py#L168-L168